imsim/ccd.py

```diff
@@ -125,16 +126,21 @@ def buildImages(self, config, base, file_num, image_num, obj_num, ignore, logger
         exp_time = base['exp_time']
         image.header['EXPTIME'] = exp_time
         image.header['DET_NAME'] = base['det_name']
-        image.header['OBSID'] = opsim_md.get('observationId', -999)
         # MJD is the midpoint of the exposure.  51444 = Jan 1, 2000, which is
         # not a real observing date.
         image.header['MJD'] = opsim_md.get('mjd', 51444)
         # MJD-OBS is the start of the exposure
         mjd_obs = opsim_md.get('observationStartMJD', 51444)
         mjd_end =  mjd_obs + exp_time/86400.
         image.header['MJD-OBS'] = mjd_obs
-        image.header['SEQNUM'] = opsim_md.get('seqnum', 0)
+        dayobs = astropy.time.Time(mjd_obs, format='mjd').strftime('%Y%m%d')
+        image.header['DAYOBS'] = dayobs
+        seqnum = opsim_md.get('seqnum', opsim_md.get('snap', 0))
```

Comment:
In `opsim_meta.py`, that calculation of `seqnum` as the nth exposure is done [here](https://github.com/LSSTDESC/imSim/blob/main/imsim/opsim_meta.py#L110).  The case where `seqnum` is not set in `opsim_meta` occurs when we are using the old instance catalog interface to get those metadata.  In that case, the earlier visits for that dayobs aren't available, so I decided to let it just equal the snap.  Maybe we should require the instance catalogs to include an seqnum field and get it from there?

Reply:
> Maybe we should require the instance catalogs to include an seqnum field and get it from there?
Yeah, I think we're more likely to produce ingest collisions if we make the default 0.  I think better to require a value.

Reply:
I added `seqnum` to the list of required entries for instance catalog inputs.  There is still a default value of `0` in ccd.py since that's needed if using `config/flat.yaml`.   To fix flat generation properly in this regard I think requires more extensive changes that would be useful here.